scapy/layers/can.py

```diff
@@ -427,9 +465,20 @@ class SignalHeader(CAN):
                                    'extended']),
         XBitField('identifier', 0, 29),
         LenField('length', None, fmt='B'),
-        ThreeBytesField('reserved', 0)
+        FlagsField('fd_flags', 0, 8, ['bit_rate_switch',
+                                      'error_state_indicator',
+                                      'fd_frame']),
+        ShortField('reserved', 0)
     ]
 
+    @classmethod
+    def dispatch_hook(cls,
+                      _pkt=None,  # type: Optional[bytes]
+                      *args,  # type: Any
+                      **kargs  # type: Any
+                      ):  # type: (...) -> Type[Packet]
+        return SignalHeader
```

Comment:
Is this useful?

Reply:
Yes, its overriding the `dispatch_hook` from `CAN`, since I derive all other functionality from there